sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -71,97 +75,248 @@ const createDomNode = (nodeName, attributes) => {
+
+    // if title is present in highlighted field, use that.
+    if (resultData.highlight !== undefined && resultData.highlight !== null) {
+        if (
+            resultData.highlight.title !== undefined &&
+            resultData.highlight.title !== null
+        ) {
+            title.innerHTML = resultData.highlight.title;
+        } else {
+            title.innerHTML = resultData.title;
+        }
+    } else {
+        title.innerHTML = resultData.title;
+    }
+
+    page_link.appendChild(title);
+    content.appendChild(page_link);
+    content.appendChild(createDomNode("br"));
+
+    // all the sections of the page.
+    let sections = resultData.inner_hits.sections;
+
+    // all the sphinx domains of the page.
+    let domains = resultData.inner_hits.domains;
+
+    if (sections !== undefined && sections !== null) {
+        for (let i = 0; i < sections.length; ++i) {
+            COUNT += 1;
+            let section_link = createDomNode("a", {
+                href: `${page_link.href}#${sections[i]._source.id}`
+            });
+            let section_title = createDomNode("span", {
+                class: "search__result__subheading"
+            });
+
+            // if section title is present in the highlighted field, use that.
+            // else use the title present in the _souce.
+            if (
+                sections[i].highlight["sections.title"] !== undefined &&
+                sections[i].highlight["sections.title"].length >= 1
+            ) {
+                section_title.innerHTML =
+                    sections[i].highlight["sections.title"][0];
+            } else {
+                section_title.innerHTML = sections[i]._source.title;
+            }
+
+            let section_content = createDomNode("p", {
+                class: "search__result__content"
+            });
+
+            // if section content is present in the highlighted field, use that.
+            // else use the content present in the _souce.
+            if (
+                sections[i].highlight["sections.content"] !== undefined &&
+                sections[i].highlight["sections.content"].length >= 1
+            ) {
+                section_content.innerHTML =
+                    "... " +
+                    sections[i].highlight["sections.content"][0] +
+                    " ...";
+            } else {
+                section_content.innerHTML =
+                    sections[i]._source.content.substring(0, 50) + " ...";
+            }
+
+            let outer_div = createDomNode("div", {
+                class: "outer_div_page_results",
+                id: "hit__" + COUNT
+            });
+            outer_div.appendChild(section_title);
+            outer_div.appendChild(section_content);
+            section_link.appendChild(outer_div);
+            content.appendChild(section_link);
+            content.appendChild(createDomNode("br", { class: "br-for-hits" }));
```

Comment:
I really dislike this way of creating HTML. Is it possible for us to use a JS template language for this rendering, instead of all being in code? I feel that would be much, much cleaner.

Reply:
I searched on the internet and found a very small and easy to use library -- [squirrelly.js](https://squirrelly.js.org/).
I have updated the PR to make use of it.
I think we can use it in other places also. So I'll be updating the code and use it where ever it proves to be effective.

Reply:
I don't have a particularly strong opinion but sphinx docs already rely on underscore which has [built-in templating](https://underscorejs.org/#template). It may not be the right fit but it could avoid an external dependency.

Reply:
@davidfischer 
Thanks.
I have removed the external dependency and used the uderscore.js for it.